components/ParameterForm.vue

```diff
@@ -0,0 +1,236 @@
+  metaData: MetaData | undefined
+  metadataFetchStatus: AsyncDataRequestStatus
+  metadataFetchError: FetchError | null
+}>();
+
+const formData = ref(
+  // Create a new object with keys set to the id values of the metaData.parameters array of objects, and all values set to refs with default values.
+  props.metaData?.parameters.reduce((accumulator, parameter) => {
+    if (parameter.parameterType !== "select" && parameter.parameterType !== "globeSelect") {
+      accumulator[parameter.id] = ref("");
+      return accumulator;
+    }
+
+    // TODO: Make default country UK after November 2024 workshop
+    if (parameter.id === "country") {
+      accumulator[parameter.id] = ref("Thailand");
+    } else {
+      // Don't set an empty value or there will be a disjoint between the select component and the formData object,
+      // since the select component will visually appear to have an option selected (the first), but the formData object will have
+      // an empty string.
+      const defaultOption = parameter.defaultOption || parameter.options[0].id;
+      accumulator[parameter.id] = ref(defaultOption);
+    }
+
+    return accumulator;
+  }, {} as { [key: string]: any }),
+);
+
+const parametersOfTypeSelect = computed(() => {
+  if (props.metaData) {
+    return props.metaData.parameters.filter(parameter => parameter.parameterType === "select");
+  } else {
+    return [];
+  }
+});
+
+const globeParameter = computed(() => {
+  if (props.metaData) {
+    return props.metaData.parameters.filter(parameter => parameter.parameterType === "globeSelect")[0];
+  } else {
+    return undefined;
+  }
+});
+
+const icon = (parameter: Parameter) => {
+  switch (parameter.id) {
+    case "country":
+      return "cilGlobeAlt";
+    case "response":
+      return "cilShieldAlt";
+    case "vaccine":
+      return "cilIndustry";
+    case "pathogen":
+      return "cilBug";
+    default:
+      return undefined;
+  }
+};
+
+const selectOptions = (parameter: Parameter) => {
+  return parameter.options.map((option: ParameterOption) => {
+    // Because the select component does not seem to honour the initial v-model value, we had to manually
+    // set the 'selected' attribute below.
+    // As a result, this list of options is recalculated each time an option is selected.
+    return { label: option.label, value: option.id, selected: option.id === formData.value![parameter.id] };
+  });
+};
+
+const optionsAreTerse = (parameter: Parameter) => {
+  const eachOptionIsASingleWord = parameter.options.filter((option) => {
+    return option.label.includes(" ");
+  }).length === 0;
+
+  return parameter.options.length <= 5 && eachOptionIsASingleWord;
+};
+
+const submitForm = () => {
+  // Not implemented yet
+};
+
+const largeScreen = ref(true);
+const breakpoint = 992; // CoreUI's "lg" breakpoint
+const setFieldSizes = () => {
+  if (window.innerWidth < breakpoint) {
+    largeScreen.value = false;
+  } else {
+    largeScreen.value = true;
+  }
+};
+
```

Comment:
This is repeated in SideBar - should really only be implemented in one place and ideally called in one place and then made available to all components. Could be put in the store and made reactive (in another ticket!)

Reply:
I'll add this to the app-store plan!